joomla/joomla-cms
# [3.10] Backport php8.1 compatibility fixes

Pull Request for Issue https://github.com/joomla/joomla-cms/pull/36041

### Summary of Changes

Take the changes from https://github.com/joomla/joomla-cms/pull/36041 and apply it to J3.10 to fix that php8.1 issues
Thanks and all the credits go to @nikosdion 

### Testing Instructions

- Install [this package](https://ci.joomla.org/artifacts/joomla/joomla-cms/3.10-dev/36083/downloads/48562/Joomla_3.10.4-dev+pr.36083-Development-Full_Package.zip) on PHP your current php version
- Install [this package](https://ci.joomla.org/artifacts/joomla/joomla-cms/3.10-dev/36083/downloads/48562/Joomla_3.10.4-dev+pr.36083-Development-Full_Package.zip) on PHP 8.1
- check the test instructions here: https://github.com/joomla/joomla-cms/pull/36041 as its a backport to 3.10

### Actual result BEFORE applying this Pull Request

issues with php 8.1

### Expected result AFTER applying this Pull Request

The known issues with php 8.1 are gone

### Documentation Changes Required

none

cc @richard67 would be great when you could take a look on your php 8.1 setup
